### PR TITLE
Update OpenTelemetry Collector to v0.60.0

### LINKED ISSUE #1993: Update the opentelemetry dependency to 0.57 (or above)
Using spanmetrics processor with many metrics causes many cache misses that prevent traces and span metrics from being sent with versions under 0.56. 

Here is the opentelemetry discussing and fixing the problem. 
https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/11149

(all other modules could be updated to 0.57 also but the fault we're facing is with processor/spanmetrics). 

Thanks in advance, 
Have a great day

### LINKED ISSUE #2155: Update go.opentelemetry.io/collector to v0.60.0
An update for `go.opentelemetry.io/collector` (version `v0.60.0`) is now available. Version `v0.55.0` is currently in use.

### PR DESCRIPTION
**Pull request by rfratto**
#### PR Description
This updates the OpenTelemetry Collector to v0.60.0. 

#### Which issue(s) this PR fixes
Fixes #2155. 
Fixes #1993. 

#### Notes to the Reviewer
This required updating the Prometheus dependency to 2.38.0.

The update was mostly smooth; after fixing all the types that moved between packages (and implementing the new UpdateMetadata method for the Prometheus update), the only real code change was to update the servicegraphprocessor's testdata, where the deprecated `instrumentationLibrarySpans` key is no longer supported and must be `scope_spans`. 

#### PR Checklist

- [x] CHANGELOG updated
- [x] Documentation added
- [x] Tests updated

### CODE REVIEW

**Comment by rfratto** [comment]
This is in a mostly good state, but I'm seeing some pretty common flakes with the operator tests where it can't find the CRDs. I'm trying to figure out what's going on there.

**Comment by rfratto** [comment]
> This is in a mostly good state, but I'm seeing some pretty common flakes with the operator tests where it sometimes can't find the CRDs. I'm trying to figure out what's going on there.

For whatever reason, one of the dependency updates changed how fast the CRDs show up. I've added in some code to wait for them to exist so the tests shouldn't flake anymore.

**Comment by rfratto** [comment]
I reran the CI a bunch of times and I feel pretty confident I fixed the flakes. I also reran just the operator tests locally 20 times, and didn't see any more flakes. I think this is ready for review now.

**Review by mapno — APPROVED** [review]
Maybe worth mentioning that the old naming `instrumentation_library...` is no longer supported (support was dropped with >v0.58). This impacts exporters and receivers. In any case, LGTM

File: docs/sources/upgrade-guide/_index.md [hunk 1]
```diff
@@ -10,10 +10,25 @@ weight: 800
 This guide describes all breaking changes that have happened in prior
 releases and how to migrate to newer versions.
 
-## Unreleased Changes
+## Unreleased changes
 
 These changes will come in a future version.
 
+### Breaking change: JSON-encocded traces from OTLP versions below 0.16.0 are no longer supported
```
Comment (karengermond) [inline docs/sources/upgrade-guide/_index.md:17]:
```suggestion
### Breaking change: JSON-encoded traces from OTLP versions below 0.16.0 are no longer supported
```

File: docs/sources/upgrade-guide/_index.md [hunk 2]
```diff
@@ -10,10 +10,25 @@ weight: 800
...
+
+Grafana Agent's OpenTelemetry Collector dependency has been updated from
+v0.55.0 to v0.60.0. OpenTelemetry Collector v0.58.0
+[removed][translation-removal] the translation for InstrumentationLibrary to
+Scope.
+
+This means that JSON-encoded traces which still use InstrumentationLibrary will
```
Comment (karengermond) [inline docs/sources/upgrade-guide/_index.md:24]:
```suggestion
This means that JSON-encoded traces that still use InstrumentationLibrary will
```

**Review by karengermond — APPROVED** [review]
LGTM

**Review by mattdurham — APPROVED** [review]
Was a bit scared when I saw the number of changed files but a good half were formatting, so LGTM

**Pull request merged**